### PR TITLE
PLANNER-2034 Use jakarta EE API for JAXB

### PR DESCRIPTION
**Pull request by rsynek**
- switches to jakarta.xml.bind:jakarta.xml.bind-api
- ecxludes any dependency that brings duplicate classes (ban-duplicated-classes enforce check must pass)

### CODE REVIEW

**Comment by rsynek** [comment]
> Although I would have thought that Quarkus have their dependencies in order, this exclusion business is a mess.

I agree. @ge0ffrey told me the Quarkus team has switching to Jakarta API artifacts on their radar, so hopefully, we can get rid of these exclusions soon.

**Comment by rsynek** [comment]
Regarding Quarkus getting rid of the jboss spec api: https://github.com/quarkusio/quarkus/issues/1394.

**Comment by rsynek** [comment]
Available PR checks passed, locally the enforcer plugin's checks passed too. Merging.

File: optaplanner-persistence/optaplanner-persistence-jpa/pom.xml [hunk 1]
```diff
@@ -53,11 +53,16 @@
...
+        <!-- Duplicates classes of jakarta.activation:jakarta.activation-api -->
         <exclusion>
           <groupId>javax.activation</groupId>
           <artifactId>javax.activation-api</artifactId>
         </exclusion>
+        <!-- Duplicates classes of jakarta.xml.bind:jakarta.xml.bind-api -->
+        <exclusion>
+          <groupId>javax.xml.bind</groupId>
+          <artifactId>jaxb-api</artifactId>
+        </exclusion>
```
Comment (ge0ffrey) [inline optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:65]: Can we avoid duplicating the exclusions by doing them in the optaplanner-build-parent?

Comment (rsynek) [reply optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:65]: Got the same question from Lukas. The answer is both yes and no:

The vast majority of the exclusions come from `io.quarkus` artifacts. If we define the exclusions in dependency management, we have to define a version of every such `io.quarkus` artifact. Effectively, we duplicate (and potentially) override the version coming from quarkus BOM.

For this reason, I would rather keep the exclusions at individual dependencies. Chances are there will be much less of them after #824 is merged.

**Pull request merged**